Topics/Tech_Stacks/FastAPI.md

```diff
@@ -0,0 +1,106 @@
+# FastAPI
+
+## Table of Contents
+### [Introduction](#introduction-1)
+### [Writing APIs](#writing-apis-1)
+### [Testing](#testing-1)
+
+
+## Introduction
+
+FastAPI is a popular web framework in Python for writing REST APIs. It is fast and easy to use, with type hints and automatic data validation. 
+
+It also auto generates documentation for your APIs with the [OpenAPI](https://www.openapis.org/) specifications and with an UI such as [Swagger UI](https://swagger.io/)
+
+You can check out FastAPI's tutorials [here](https://fastapi.tiangolo.com/tutorial/) at their own website with instructions on how to install FastAPI and how to get started with your first REST API using FastAPI
+
+Or if you wish to browse their documentation at your leisure, click [here](https://fastapi.tiangolo.com/)
+
+
+## Writing APIs
+
+The FastAPI tutorial on writing your first API from [here](https://fastapi.tiangolo.com/tutorial/first-steps/) is a very good starting point. It shows that all your API files will look something like this:
+
+``` {python}
+from fastapi import FastAPI
+
+app = FastAPI()
+
+# your APIs here
+@app.get("/test")
+def test():
+  return {"test message": "hello world"}
+
+```
+
+However, it is highly likely that not all your APIs are going to be in the same file. For example, take the above as your main.py. You are likely not going to put all your APIs there, so you may create a file called
```

Comment:
It may be worth briefly elaborating on this point by showing what the file structure for an API could look like, with its folders and files inside it, like:

```
api
├── user
│   ├── user.py
├── more
│   ├── more.py
└── main.py
```